### PR TITLE
Fixes evaluation console, drops missing values before training and fixes time elapsed in training

### LINKED ISSUE #1918: Confusing console output on drawing_classifier.evaluate(...)
```python
In [17]: len(test_data)
Out[17]: 143

In [18]: _ = model.evaluate(test_data)
Predicting 144/143

In [19]: len(longer_test_data)
Out[19]: 429

In [20]: _ = model.evaluate(longer_test_data)
Predicting 257/429
```
The output from the first call to `evaluate(...)` is really the most confusing. How is it predicting one more examples than was passed in? This seems like an off by one bug.

The output from the second call to `evaluate(...)` makes it seem like it never finishes. The last/only output makes it seem like it's only about half done. The final output should probably be `Predicting X/X` where `X` is the number of examples passed in.

Both calls to evaluate finish quickly. I would recommend only outputting progress info if it's been at least 30 seconds.

### LINKED ISSUE #1936: Bad Error Message when training with None Types
If some of the images are `None`, the drawing classifier give a poor error message:
```
In [1]: import turicreate as tc

In [2]: from random import randint

In [3]: data = tc.load_sframe('./quickdraw/sframes/bitmap.sframe/')

In [4]: data['drawing'] = data['drawing'].apply(lambda x: None if randint(1, 20) == 10
                                                                                                              else x)

In [5]: assert len(data['drawing'].dropna()) < len(data['drawing'])

In [6]: tc.drawing_classifier.create(data, feature='drawing', target='label')
PROGRESS: Creating a validation set from 5 percent of training data. This may take a while.
          You can set ``validation_set=None`` to disable validation tracking.

+------------------------+------------------------+------------------------+------------------------+------------------------+
| Iteration              | Training Loss          | Training Accuracy      | Validation Accuracy    | Elapsed Time (seconds) |
+------------------------+------------------------+------------------------+------------------------+------------------------+
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-da952d119295> in <module>()
----> 1 tc.drawing_classifier.create(data, feature='drawing', target='label')

/Users/tobyroseman/virts/rcTesting/lib/python2.7/site-packages/turicreate/toolkits/drawing_classifier/drawing_classifier.pyc in create(input_dataset, target, feature, validation_set, warm_start, batch_size, max_iterations, verbose)
    276             accuracy_metric.update(batch_label, outputs)
    277 
--> 278     for train_batch in train_loader:
    279         train_batch_data, train_batch_label = get_data_and_label_from_batch(train_batch)
    280         with _autograd.record():

/Users/tobyroseman/virts/rcTesting/lib/python2.7/site-packages/turicreate/toolkits/drawing_classifier/_sframe_loader.pyc in next(self)
    137 
    138     def next(self):
--> 139         return self._next()
    140 
    141     @property

/Users/tobyroseman/virts/rcTesting/lib/python2.7/site-packages/turicreate/toolkits/drawing_classifier/_sframe_loader.pyc in _next(self)
    157         for b in range(self.batch_size):
    158             try:
--> 159                 row = next(self.data_source)
    160             except StopIteration:
    161                 if b == 0:

/Users/tobyroseman/virts/rcTesting/lib/python2.7/site-packages/turicreate/toolkits/drawing_classifier/_sframe_loader.pyc in next(self)
     38 
     39     def next(self):
---> 40         return self._next()
     41 
     42     def _next(self):

/Users/tobyroseman/virts/rcTesting/lib/python2.7/site-packages/turicreate/toolkits/drawing_classifier/_sframe_loader.pyc in _next(self)
     59         is_stroke_input = (not isinstance(drawing_feature, _tc.Image))
     60         if is_stroke_input:
---> 61             pixels_01 = drawing_feature.pixel_data.reshape(
     62                 1, bitmap_width, bitmap_height) / 255.
     63         else:

AttributeError: 'NoneType' object has no attribute 'pixel_data'
```
I would never guess that this mean my data has `None`s in it. 

At the very least we should give a clear error message. It would be better to drop the rows with `None` in them and give a warning about it.

### PR DESCRIPTION
**Pull request by shreyajain17**
fixes #1918
fixes #1936

### CODE REVIEW

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 1]
```diff
@@ -602,7 +614,7 @@ def _predict_with_probabilities(self, input_dataset, batch_size=64,
                 if verbose and (dataset_size >= 5
                     and cur_time > last_time + 30 or done):
                     print('Predicting {cur_n:{width}d}/{max_n:{width}d}'.format(
-                        cur_n = index + 1 if not done else index,
+                        cur_n = index + 1 if index + 1 > dataset_size - 1 else index,
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: I'm having a hard time understanding this line change.

Is `done` not actually getting set to `True` when things are done?

Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: So I had made this change in my earlier PR with not done in my earlier PR but that hadn't fixed this issue. If you see index is being incremented by split_length and then checks if it is equal to dataset_size - 1. So basically index skips over the dataset_size - 1 value for done to be true. But I am not fixing it correctly I should have just make checked if index > dateset_size -1 and kept this code as it is. Does this make sense?

Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: That sounds good. Let's not change this line and let's just make sure that `done` is getting set correctly.

Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Yes, got it. I will change it.

Comment (nickjong) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: This code reads weirdly to me. AFAICT, `index` exists only to track how many instances have been processed. I would be in favor of just renaming it `cur_n` directly. I also think we can just remove the variable `done` entirely. It looks to me like the current computation of `done` is incorrect, it should be:
```
if cur_n >= dataset_size:
  cur_n = dataset_size
```
And then instead of testing `done` you can just test `cur_n >= dataset_size`.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 2]
```diff
@@ -197,6 +197,18 @@ def create(input_dataset, target, feature=None, validation_set='auto',
         raise TypeError("Unrecognized type for 'validation_set'."
             + validation_set_corrective_string)
 
+
+    original_dataset_size = len(dataset)
+    dataset = dataset.dropna(columns=feature)
+    if original_dataset_size != len(dataset):
+        print("Dropping " + str(original_dataset_size - len(dataset)) + " rows from the training dataset.")
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Could we indicate in the message why we're dropping the user's data?

Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Dropping x rows for containing None values in the training dataset?

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 3]
```diff
@@ -197,6 +197,18 @@ def create(input_dataset, target, feature=None, validation_set='auto',
         raise TypeError("Unrecognized type for 'validation_set'."
             + validation_set_corrective_string)
 
+
+    original_dataset_size = len(dataset)
+    dataset = dataset.dropna(columns=feature)
```
Comment (srikris) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Won't this mutate the original dataset? I think we should not change the user data.

Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: I don't see how. `dropna` returns a new SFrame, doesn't change things in place.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 4]
```diff
@@ -197,6 +197,18 @@ def create(input_dataset, target, feature=None, validation_set='auto',
         raise TypeError("Unrecognized type for 'validation_set'."
             + validation_set_corrective_string)
 
+
+    original_dataset_size = len(dataset)
```
[the remark this answers is not in the document]
Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: I can do that

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 3 above)
Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: It doesn't mutate the data

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 5]
```diff
@@ -44,6 +44,17 @@ def _raise_error_if_not_drawing_classifier_input_sframe(
     if len(dataset) == 0:
         raise _ToolkitError("Input Dataset is empty!")
 
+def drop_missing_values(dataset, feature, is_train):
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: This method should be hidden. Please prefix method name with a `_`.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 6]
```diff
@@ -44,6 +44,17 @@ def _raise_error_if_not_drawing_classifier_input_sframe(
...
+    original_dataset_size = len(dataset)
+    dataset = dataset.dropna(columns=feature)
+    if is_train:
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Nit: this if/else should probably go under the if statement below, only if we're actually going to print something. Then it can also be a one liner:
```
if original_dataset_size != len(dataset):
  dataset_type = "train" if is_train else "validation"
  ...
```

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 7]
```diff
@@ -594,15 +608,13 @@ def _predict_with_probabilities(self, input_dataset, batch_size=64,
...
 
                 cur_time = _time.time()
                 # Do not print progress if only a few samples are predicted
                 if verbose and (dataset_size >= 5
-                    and cur_time > last_time + 30 or done):
-                    print('Predicting {cur_n:{width}d}/{max_n:{width}d}'.format(
-                        cur_n = index + 1 if not done else index,
+                    and cur_time > last_time + 30 or index >= dataset_size ):
+                    print('Predicting {num_processed:{width}d}/{max_n:{width}d}'.format(
+                        num_processed = index,
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Has this change been testing? I'm not certain but if it's possible for this to be true `index > dataset_size - 1`, then we could be reporting that we processed more example than there are in the dataset.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 8]
```diff
@@ -561,7 +576,6 @@ def _predict_with_probabilities(self, input_dataset, batch_size=64,
 
         index = 0
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Can we just rename `index` to `num_processed`?

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 9]
```diff
@@ -297,7 +313,6 @@ def compute_accuracy(accuracy_metric, batch_loader):
         trainer.step(train_batch.data[0].shape[0])
         # calculate training metrics
         train_loss = loss.mean().asscalar()
-        train_time = _time.time() - start_time
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Why this change and the next two changes? Are you fixing another issue?

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 10]
```diff
@@ -594,15 +608,13 @@ def _predict_with_probabilities(self, input_dataset, batch_size=64,
...
                 cur_time = _time.time()
                 # Do not print progress if only a few samples are predicted
                 if verbose and (dataset_size >= 5
-                    and cur_time > last_time + 30 or done):
-                    print('Predicting {cur_n:{width}d}/{max_n:{width}d}'.format(
-                        cur_n = index + 1 if not done else index,
+                    and cur_time > last_time + 30 or index >= dataset_size ):
+                    print('Predicting {num_processed:{width}d}/{max_n:{width}d}'.format(
+                        num_processed = index,
                         max_n = dataset_size,
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: In this print statement, let's either use "num_processed" and "max_processed" or "cur_n" and "max_n"

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 7 above)
Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: This works. 

```
model.evaluate(data)
Predicting 256/256
```

```
In [13]: model.evaluate(data)
Predicting  256/7237
Predicting 7237/7237
```

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 5 above)
Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Okay

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 9 above)
Comment (shreyajain17) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: In one of the previous PRs, I was suspicious the time elapsed is incorrectly calculated. The time printing earlier was for one mini batch of the epoch, this change correctly calculates time taken per epoch.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 11]
```diff
@@ -197,6 +205,11 @@ def create(input_dataset, target, feature=None, validation_set='auto',
         raise TypeError("Unrecognized type for 'validation_set'."
             + validation_set_corrective_string)
 
+    dataset = _drop_missing_values(dataset, feature, is_train =True)
+
+    if validation_set is not None:
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Nit: `validation_dataset` is always an (possibly empty) `SFrame` at this point, right? Could we just unconditionally do line 211?
